bindings/wasm/src/tx_types/contract/contract_matching.rs

```diff
@@ -38,7 +39,17 @@ impl ContractMatchingBuilder {
         maker: Vec<JsValue>,
         fee: String,
         fee_token: u16,
+        contract_prices: Vec<JsValue>,
+        margin_prices: Vec<JsValue>,
     ) -> Result<ContractMatchingBuilder, JsValue> {
+        let contract_prices = contract_prices
+            .iter()
+            .map(|p| serde_wasm_bindgen::from_value(p.clone()).unwrap())
+            .collect::<Vec<InnerContractPrice>>();
+        let margin_prices = margin_prices
+            .iter()
+            .map(|p| serde_wasm_bindgen::from_value(p.clone()).unwrap())
+            .collect::<Vec<InnerSpotPriceInfo>>();
```

Comment:
collect with Result, `let new: Result<Vec<T>, E> = v.into_iter().collect()`, and use `map_err`